### PR TITLE
Support OutlierDetection for grpc proxyless

### LINKED ISSUE #48022: OutlierDetection Support for Grpc-Xds proxyless
**Describe the feature request**
istio grpc-proxyless support for outlierDetection

Currently istio supports outlierDetection feature for only envoy proxy and being one of the performant features of istio which we to use in our production setup, but this feature is not supported for grpc-xds proxyless setup.

Grpc-xds supports only 2 algorithms [xds-grfc](https://github.com/grpc/proposal/blob/master/A50-xds-outlier-detection.md)

1. Failure Percent
2. Success Rate 

Envoy's [outlierDetection](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier#outlier-detection) also the algorithm support but istio does not provide any way to configure these. In order to unblock we tried to use EnvoyFilter to configure those algorithm parameters. We were able to do this in presence of sidecar Envoy proxy as istio does provide the patching of EnvoyFilterConfig  over the [clusterConfig](https://github.com/istio/istio/blob/141104caec71593378094d789a47e1da7979b57b/pilot/pkg/networking/core/v1alpha3/cluster_builder.go), but this patching was not provider for [grpc-proxyless](https://github.com/istio/istio/blob/141104caec71593378094d789a47e1da7979b57b/pilot/pkg/networking/grpcgen/cds.go)

hence a better solution would be if istio supports OutlierDetection for grpc-proxyless where we wont need to patch envoyFilter config as a workaround and configure outlierDetection from DestinationRule itself.

**Describe alternatives you've considered**
As an alternative we tried to use EnvoyFilter which patched the its config along the clusterConfig generated after applying destinationRule.
discussion mention here : [Alternative](https://github.com/istio/istio/issues/47927)
**Affected product area (please put an X in all that apply)**

[ ] Ambient
[x] Docs
[ ] Dual Stack
[ ] Installation
[x] Networking
[x] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

### PR DESCRIPTION
**Pull request by aditya-32**
Closes #48022 
related [istio/api](https://github.com/istio/api/issues/3001)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,6 +5,8 @@ go 1.21
 // Client-go does not handle different versions of mergo due to some breaking changes - use the matching version
 replace github.com/imdario/mergo => github.com/imdario/mergo v0.3.5
 
+replace istio.io/api => github.com/aditya-32/api v1.0.0
```
Comment (zirain) [inline go.mod:8]: why need this?

File: pilot/pkg/networking/grpcgen/cds.go [hunk 2]
```diff
@@ -206,9 +206,58 @@ func (b *clusterBuilder) applyTrafficPolicy(c *cluster.Cluster, trafficPolicy *n
 	}
 	b.applyTLS(c, trafficPolicy)
 	b.applyLoadBalancing(c, trafficPolicy)
+	b.applyOutlierDetection(c, trafficPolicy)
 	// TODO status or log when unsupported features are included
 }
 
+func (b *clusterBuilder) applyOutlierDetection(c *cluster.Cluster, policy *networking.TrafficPolicy) {
```
Comment (anuragagarwal561994) [inline pilot/pkg/networking/grpcgen/cds.go:213]: We should also implement this for the envoy code flow

**Comment by anuragagarwal561994** [comment]
Closes https://github.com/istio/istio/issues/40785
Closes https://github.com/istio/istio/issues/19052

**Pull request closed without merging**